File: 0927/tried/G.py
T = int(input())
def gcd(a, b):
    return b if a%b == 0 else gcd(b, a%b)
mem = {}
for _ in range(T):
    n, m = map(int, input().split())
    n, m = (n, m) if n<=m else (m, n)
    
    t = (n,m)
    if t in mem:
        print(mem[t])
        continue

    pts = [(x, y) for y in range(m) for x in range(n)]
    tot = 0
    for p1 in pts:
        for p2 in pts:
            if p1 == p2:
                break
            dx = p1[0] - p2[0]
            dy = p1[1] - p2[1]
            g = 1 if dx == 0 or dy == 0 else gcd(abs(dx), abs(dy))
            dx = dx//g
            dy = dy//g
            if dx == 0:
                tot += n*m - n
                continue
            if dy == 0:
                tot += n*m - m
                continue
            rml = p1[1]//dy + 1
            rmr = (m-1 - p1[1])//dy
            if dx < 0:
                rml = min(rml, -(n-1 - p1[0])//dx + 1)
                rmr = min(rmr, -(p1[0])//dx)
            else:
                rmr = min(rmr, (n-1 - p1[0])//dx)
                rml = min(rml, (p1[0])//dx + 1)
            add = n*m - rmr - rml
            tot += add

    tot = tot//3
    mem[(n,m)] = tot
    print(tot)

File: 0927/tried/G.java
import java.util.*;
import java.io.*;
public class G {
    int gcd(int a, int b) {
        return (a%b == 0)? b: gcd(b, a%b);
    }
    void solve(BufferedReader in) throws Exception {
        int T = toInt(in.readLine());
    for (int t = 0; t<T; t++){
        int[] xx = toInts(in.readLine());
        int n = xx[0], m = xx[1];

        int sum = 0;
        for(int p1 = 0; p1<n*m; p1++) {
            int p10 = p1%n;
            int p11 = p1/n;
            for (int p2 = 0; p2<p1; p2++) {
                int p20 = p2%n;
                int p21 = p2/n;

                int dx = p10 - p20;
                int dy = p11 - p21;
                if (dx == 0){
                    sum += n*m - m;
                    continue;
                }
                if (dy == 0) {
                    sum += n*m - n;
                    continue;
                }
                int g = gcd(Math.abs(dx), Math.abs(dy));
                int rml = p11/dy + 1;
                int rmr = (n-1 - p11)/dy;
                if (dx < 0){
                    rml = Math.min(rml, -(m-1 - p10)/dx + 1);
                    rmr = Math.min(rmr, -(p10)/dx);
                }else{
                    rmr = Math.min(rmr, (m-1 - p10)/dx);
                    rml = Math.min(rml, (p10)/dx + 1);
                }
                sum += n*m - rmr - rml;
            }
        }
        System.out.println(sum/3);
        
    }
    }
    int toInt(String s) {return Integer.parseInt(s);}
    int[] toInts(String s) {
        String[] a = s.split(" ");
        int[] o = new int[a.length];
        for(int i = 0; i<a.length; i++) o[i] = toInt(a[i]);
        return o;
    }
    void e(Object o) {
        System.err.println(o);
    }
    public static void main(String[] args) throws Exception{
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        (new G()).solve(in);
    }
}
